Collapsable banners for survey annoucement

Wineshuga opened the pull request:
### Description

This pull request introduces a new banner feature across the website. The changes involve adding the banner's HTML structure, styling, and functionality.
I found this open issue (#2625), and it looked like something I could do, so I decided to give it a try. I would appreciate some feedback. 

### Banner Feature Implementation:
* Added a new banner in `layouts/banner.html` with a survey link and a close button.
* Integrated the banner into the site's header by rendering it in `layouts/header.html`.
* Implemented a `hideBanner` JavaScript function in `layouts/footer.html` to allow users to close the banner.

### Screenshot

<img width="958" alt="image" src="https://github.com/user-attachments/assets/71850cc6-79f4-46f4-aaf9-e7ba786dbf3b" />

ArthurSens commented:
Hey @Wineshuga, thanks for raising the PR! I'm not a front-end engineer, but the preview showed the behavior I wanted to see ❤️.

There are some things we need to clean up first, though:
* It looks like you accidentally added a few extra commits to this PR. Could you please make sure only the relevant commits are here?
* I'm seeing a lot of unrelated changes, probably because of some automatic linting configuration in your IDE. Could you revert them and commit only what's relevant to the PR?

It looks like @isabellepinheiro raised https://github.com/prometheus/docs/pull/2629, but the banner is only visible on the index page. Isabelle, since you know more about front-end code than I do, could you review this PR here? Mostly about styling?

Wineshuga commented:
Thank you, @ArthurSens, for the helpful feedback. I've removed the unrelated commits that were mistakenly included and also cleaned up the changes caused by my IDE’s automatic linting. Please let me know if there's anything else you'd like me to adjust.

ArthurSens commented:
Thank you, let's ship it!

ArthurSens commented:
Urgh, for some reason it didn't work well when deployed. I'm looking for extra help here

<img width="1191" alt="image" src="https://github.com/user-attachments/assets/246ff257-e547-4098-9c77-a56c0e248091" />
